graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,193 @@
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="LRO.gif" alt="The status monitor LRO flow"/>
+</p>
+<!-- markdownlint-enable MD033 -->
+
+The RELO pattern is the preferred pattern for long running operations and should be
+used wherever possible. The pattern avoids complexity, and consistent resource
+presentation makes things simpler for our users and tooling chain.
+
+In general, Microsoft Graph API guidelines for LRO follow [Microsoft REST API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations).
+
+There are some deviations from the base guidelines where Microsoft Graph API standards require that you do one of the following:
+
+- For RELO pattern you should return the Content-Location header that indicates the location of the resource.
+  - The API response says the targeted resource is being created by returning a 201 status code and the resource URI is provided in the Content-Location header, but the response indicates that the request is not completed by including "Provisioning" status.
+
+- For LRO pattern you should return the Location header that indicates the location of a new stepwise operation resource.
+  - The API response says the operation resource is being created at the URL provided in the Location header and indicates that the request is not completed by including a 202 status code.
+  - Microsoft Graph doesn’t allow tenant wide operation resources therefore stepwise operations is often modeled as a navigation property on the target resource.
+
+## When to use this pattern
+
+Any API call that is expected to take longer than 1 second in the 99th percentile should use the long running operations pattern.
+
+How do you select which flavor of LRO pattern to use? An API designer can follow these
+heuristics:
+
+1.  If a service can create a resource with a minimal latency and continue
+    updating its status according to the well-defined and stable state
+    transition model until completion, then the RELO model is the best choice.
+
+2.  Otherwise, a service should follow the Stepwise Operation pattern. 
+ 
+
+## Issues and considerations
+
+- One or more API consumers MUST be able to monitor and operate on the same resource at the same time.
+
+- The state of the system SHOULD always be discoverable and testable. Clients
+    SHOULD be able to determine the system state even if the operation tracking
+    resource is no longer active. Clients MAY issue a GET on some resource to
+    determine the state of a long running operation.
+
+-  The long running operations pattern SHOULD work for clients looking to "fire and forget"
+    and for clients looking to actively monitor and act upon results.
+
+-  The long running operations pattern might be supplemented by the [change notification
+    pattern](change-notification.md).
+
+-  Cancellation of a long running operation does not explicitly mean a rollback. On a per API-defined case, it
```

Comment:
Can you provide examples of what this looks like for RELO?  
I have no idea what the request looks like to cancel RELO.  The Microsoft API Guidelines describe cancellation for LRO but not for RELO...

Reply:
Provided